src/_includes/tools/dart-compile-js-options.md

```diff
@@ -0,0 +1,112 @@
+#### Options {#prod-compile-options}
+
+Dart supports multiple options to customize code compilation.
+
+* [Basic options](#basic-options)
+* [Path and environment options](#path-and-environment-options)
+* [Display options](#display-options)
+* [Analysis options](#analysis-options)
+
+###### Basic options
+
+Common options include:
+
+`-o <file>` or `--output=<file>`
+: Generates the output into `<file>`. 
+  If not specified, the output goes in a file named `out.js`.
+
+`--enable-asserts`
+: Enables assertion checking.
+
+`-O{0|1|2|3|4}`
+: Controls optimizations to reduce file size and
+  improve code performance.
+  To learn more about these optimizations, 
+  run `dart compile js -hv`.
+
+  * `-O0`: Disables many optimizations.
+  * `-O1`: Enables default optimizations.
+  * `-O2`: Enables `-O1` optimizations, plus additional ones
+    (such as minification) that respect the language semantics and
+    are safe for all programs.
+    {{site.alert.note}}
+      With `-O2`, string representations of types are no longer the same as
+      those in the Dart VM and when compiled with [dartdevc][].
+    {{site.alert.end}}
+  * `-O3`: Enables `-O2` optimizations, plus omits implicit type checks.
+    {{site.alert.warning}}
+      Omitting type checks can cause your app to crash due to type errors.
+      Before using `-O3`, **test using `-O2`** to ensure that your app
+      **never** throws a subtype of `Error` (such as `TypeError`).
+    {{site.alert.end}}
+  * `-O4`: Enables more aggressive optimizations than `-O3`,
+    but with the same assumptions.
+    {{site.alert.warning}}
+      The `-O4` optimizations are susceptible to variations in input data.
+      Before relying on `-O4`, **test for edge cases in user input**.
+    {{site.alert.end}}
+
+`--no-source-maps`
+: Do not generate a source map file.
+
+`-h` or `--help`
+: Displays help. To get information about all options, use `-hv`.
+
+
+###### Path and environment options
+
+Some other handy options include:
+
+`--packages=<path>`
+: Specifies the path to the package resolution configuration file.
+  For more information, see
+  [Dart Package Configuration File.][]
+
+`-D<flag>=<value>`
+: Defines an environment variable.
```

Comment:
We have to have a doc on this that can be linked to.

I'm annoyed this page https://api.dart.dev/stable/2.18.0/dart-core/int/int.fromEnvironment.html – and others don't link to something like that!